### ISSUE TITLE
Tor + Opera VPN = Connection problems

### ISSUE THREAD

**Issue by Backroom-Studio** [andy-portmen/tor-bundle#1]
IDK for sure if this is a bug in the extension or not. But I figured I would let you know either way. So, when I turn off Tor using your extension Opera will no longer connect to any websites. It shows an error page that reports no Internet connection. My connection works in other browsers and will work in Opera if I turn on Tor or if I turn on the built in VPN feature., So I figure it has something to do with the proxy settings not going back to default? idk for sure., Here is what I did exactly if you want to try to recreate the bug.

1.) install extension + bundle + client (with vpn off)
2.) connect to tor using extension and verify tor (with vpn off)
*Verification good & extension reports tor as ON
3.) turn on vpn & verify Tor. (with vpn on)
*Verification failed - extension reports tor as OFF
4.) connect to Tor using extension & verify tor (with vpn on)
*Verification failed & extension reports tor as ON
5.) turn off vpn & verify tor (with vpn off)
*Verification good & extension reports tor as ON
6.) turn off tor using extension (with vpn off)
*No connection error
7.) turn on either vpn or tor (but not both)
*connection works

so... ummm, yeah.  Let me know if you can re-create the issue or if my report has helped you at all..
backroom-studio@gmx.com

**Comment by andy-portmen** [comment]
So basically Opera VPN is a proxy server not a VPN, but it overwrites the default proxy that is set by the extension. So whenever Opera VPN is on, verification will fail.

I am not following the step 6 though. You mean even if we forget about Opera's internal VPN, turning the extension off, is not releasing the proxy setting?

**Comment by Backroom-Studio** [comment]
yes the etension isn't releasing the proxy setting., I only get connection back by disabling the extension...  Even now, if I simply enable the extension, i lose my connection unless TOR is turned on in the extension., If Tor is set to off I have no connection.  Disable the extension and connection works again.

The only thing I could imagine is that I caused some kind of bug by turning ON the VPN while the extension actively had control of the proxy....  So the VPN hijacked control of the proxy from the extension with caused some kind of problem.



eric.zollman@gmx.com
 
From: andy portmen
Date: 2017-07-12 07:39
To: andy-portmen/tor-bundle
CC: Backroom Studio; Author
Subject: Re: [andy-portmen/tor-bundle] Tor + Opera VPN = Connection problems (#1)
So basically Opera VPN is a proxy server not a VPN, but it overwrites the default proxy that is set by the extension. So whenever Opera VPN is on, verification will fail.
I am not following the step 6 though. You mean even if we forget about Opera's internal VPN, turning the extension off, is not releasing the proxy setting?

**Comment by andy-portmen** [comment]
Please wait for 0.1.1 release. It should fix the issue. If not please reopen
